### ISSUE TITLE
Build fails: Can't copy resources

### ISSUE THREAD

**Issue by gerrymclarnon** [microsoft/vscode-remote-release#2051]
Adding some resources (intermittently) causes a build error, "Failed to copy full contents",  on  Windows 10.

**Comment by Chuxel** [comment]
Can you include a full log from the "Dev Containers" terminal window when this happens? (View > Terminal, Select Dev Containers from the dropdown to the right.)

**Comment by gerrymclarnon** [comment]
I couldn't find a way to view the Dev Containers terminal unless I rebuilt my container. I've attached the log but I don't see anything there of much use. I've also attached the output from the maven build that is causing the problem, I suspect it is something to do with dodgy Windows 10 volume mounts in Docker.
Thanks for your help.

[dev-containers.txt](https://github.com/microsoft/vscode-remote-release/files/3977935/dev-containers.txt)
[maven-output.txt](https://github.com/microsoft/vscode-remote-release/files/3977936/maven-output.txt)

**Comment by Chuxel** [comment]
What version of Docker Desktop are you on?  I'm seeing some inidcation of issues with bind mounts in 2.1.7.0: https://github.com/docker/for-win/issues/5318

**Comment by gerrymclarnon** [comment]
I'm on 2.1.0.5.

**Comment by Chuxel** [comment]
If you do the same thing outside of a container do you see the same behavior?  The other possibility here is actually a Maven bug. The sample recently updated from 3.6.1 to 3.6.3.

Also, is this happening after your machine was asleep or you moved between wifi networks? Docker uses windows (CIFS) shares for bind mounts so sometimes this can affect them.  (This will change with the upcoming WSL2 implementation, but that doesn't help near term.)

**Comment by gerrymclarnon** [comment]
FWIW the same maven build works fine on a Mac inside and outside of a container. I have to use a container on Windows due to a locked down environment. :-(

The issues doesn't appear to be related to sleeping or wifi, occurs intermittently when I'm at the machine on a hard wired connection.

Thanks...

**Comment by Chuxel** [comment]
Hmmm - This sounds very similar to this issue that this maven plugin appears to use: https://issues.apache.org/jira/browse/IO-544

...and this is a related issue with the exact error...

https://issues.apache.org/jira/browse/IO-443

It looks like there is some sort of issue with Java and flushing SMB mounted files -- which is this case. There's then an unneeded check that is failing.

There also are some closed issues that sound like this that tie back to the bug:

https://issues.apache.org/jira/browse/MRESOURCES-48
https://issues.apache.org/jira/browse/MRESOURCES-51

It might  be worth raising an issue with Maven.

**Comment by gerrymclarnon** [comment]
I've tried 3 different JDK & Maven version combinations and all suffer from the same problem.

I've also tried developing an Angular app in a dev container setup and so far haven't experienced any problems, so that seems to support the issue being related to Maven.

Perhaps my best chance of Java development with VS Code on Windows is if the locked down development environment is updated or unlocked.

Thanks for your help.
